pkg/errors/errors.go

```diff
@@ -0,0 +1,7 @@
+package errors
```

Comment:
Let's not make this a new package. I think it's just fine defining this in the `caches.go` file and use it via `generator.ErrDomainConflict` in the reconciler. I'm personally not a fan of centralized errors like this and would rather keep them close to their implementation.